projects/components/src/color-picker/color-picker.component.ts

```diff
@@ -78,7 +82,7 @@ export class ColorPickerComponent implements ControlValueAccessor, OnChanges {
   }
 
   public selectColor(color: string): void {
-    if (this.disabled) {
+    if (this.disabled || (this.disableDeselection && color === this.selected)) {
```

Comment:
nit - the special casing for deselecting happens twice - here and on 89. Would be a bit nicer to resolve the selection, then handle if the result is allowed. Would be even nicer to prevent deselecting in the first place - disabling the interaction on the current color.

Reply:
The second part here seems like the important one btw. Not handling this in code but in UX.